include/ensmallen_bits/cmaes/cmaes_impl.hpp

```diff
@@ -192,27 +198,31 @@ typename MatType::elem_type CMAES<SelectionPolicyType>::Optimize(
 
     // Calculate the objective function.
     currentObjective = selectionPolicy.Select(function, batchSize,
-        mPosition[idx1], callbacks...);
+      transformationPolicy.Transform(mPosition[idx1]), callbacks...);
 
     // Update best parameters.
     if (currentObjective < overallObjective)
     {
       overallObjective = currentObjective;
       iterate = mPosition[idx1];
 
-      terminate |= Callback::StepTaken(*this, function, iterate, callbacks...);
+      transformedIterate = transformationPolicy.Transform(iterate);
+      terminate |= Callback::StepTaken(*this, function, 
+        transformedIterate, callbacks...);
     }
 
     // Update Step Size.
     if (iterate.n_rows > iterate.n_cols)
     {
       ps[idx1] = (1 - cs) * ps[idx0] + std::sqrt(
-          cs * (2 - cs) * muEffective) * covLower.t() * step;
```

Comment:
If the goal is the compute `C^{-1/2}`, it should suffice to use `inv(covLower)`: https://math.stackexchange.com/questions/1230051/inverse-square-root-of-matrix

I think Armadillo will solve the system more quickly too if you specify `trimatl()` (since it is a lower triangular matrix): `inv(trimatl(covLower))`.

I haven't checked this exactly, but it should at least be in the right direction.  You should verify my comment here instead of trusting it to be correct... :smile:

Reply:
In Nikolaus Hansen's [The CMA Evolution Strategy: A Tutorial](https://arxiv.org/pdf/1604.00772v1.pdf) `C^{-1/2}` is defined to be `B * D^{-1} * B^T` which isn't always equal to `covLower = B * D^{-1}`

Reply:
I understand and agree. I haven't had much time to think about this either but I'll try to figure something out soon.